src/lib/hookify/hookedSelectors.ts

```diff
@@ -0,0 +1,229 @@
+import { useSyncExternalStore } from 'react';
+
+import { Action, Dispatch, EnhancedStore, UnknownAction } from '@reduxjs/toolkit';
+import { QueryClient, QueryObserver, useQuery } from '@tanstack/react-query';
+import { isFunction } from 'lodash';
+
+import { getUseBaseQuery } from './useBaseQuery';
+import hookifyHooks, { useSyncExternalStoreHf } from './vanillaHooks';
+
+type HookSub<Return> = (val: Return) => void;
+
+// wraps a function in vanilla hooks and returns an interface that lets users
+// set the arguments with call() and subscribe to all return values with subscribe()
+type Hookified<Return, Args extends any[]> = {
+  // callback whenever the return value changes for any reason
+  subscribe: (handle: HookSub<Return>) => () => void;
+
+  // will return undefined if never called or destroyed
+  getLatestValue: () => Return | undefined;
+
+  // sets the arguments and returns the first return value after running with these args
+  // further triggered changes available via subscribe()
+  call: (...args: Args) => Return;
```

Comment:
is `call` only expected to be used once per `hookify(...))`?

Reply:
Nope, you can call it as many times as you want. Each call saves the new set of arguments as the correct ones and runs the function once (which may trigger further runs in useEffects and such).